core/modules/nfv_ctrl.cc

```diff
@@ -139,8 +151,15 @@ int NFVCtrl::NotifyRCoreToRest(cpu_core_t core_id, int q_id) {
   }
 
   cpu_core_t down = bess::ctrl::sw_q_state[q_id]->down_core_id;
-  bess::ctrl::nfv_rcores[down]->RemoveQueue(bess::ctrl::sw_q[q_id]);
-  bess::ctrl::rcore_state[down] = true; // reset so that it can be assigned later
+  if (down == DEFAULT_NFVCTRL_CORE_ID) {
+    // Queue has been assigned to nfv_ctrl for dumping.
+    std::vector<struct llring*>::iterator it = std::find(sw_q_to_drop_.begin(), sw_q_to_drop_.end(), bess::ctrl::sw_q[q_id]);
+    assert (it != sw_q_to_drop_.end());
+    sw_q_to_drop_.erase(it);
```

Comment:
Removing a queue to std::vector can cause race condiions. I will do a quick fix on it.